src/kvstore/NebulaStore.cpp

```diff
@@ -820,6 +822,29 @@ int32_t NebulaStore::allLeader(std::unordered_map<GraphSpaceID,
     return count;
 }
 
+// Only for testing
+bool NebulaStore::waitNLeadersOnSpace(GraphSpaceID spaceId, uint64_t leaders) const {
+    while (true) {
+        uint64_t count = 0;
+        {
+            folly::RWSpinLock::ReadHolder rh(&lock_);
+            auto space = spaces_.find(spaceId);
+            if (space == spaces_.end()) {
+                return false;
+            }
+            for (const auto& partIt : space->second->parts_) {
+                if (partIt.second->isLeader()) {
+                    ++count;
+                }
+            }
+        }
+        if (count == leaders) {
+            return true;
+        }
+        ::sleep(FLAGS_raft_heartbeat_interval_secs * 2);
+    }
+}
+
```

Comment:
Thank for your efforts!
But I don't think that's a good solution,  This method should not appear here if only for testing.
Could you refer to method `waitUntilAllElected` if you wish?

Reply:
Good point.